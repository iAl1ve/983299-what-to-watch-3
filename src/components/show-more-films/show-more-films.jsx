import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getFilmsToRender} from '../../reducer/data/selectors';
import {getFilmsToShowCount} from '../../reducer/app-status/selectors';
import {ActionCreators} from '../../reducer/app-status/app-status';

function ShowMoreFilms({onMoreFilmsButtonClick, filmsToRender, filmsToShowCount}) {
  return (
    <div style={filmsToShowCount > filmsToRender.length ? {display: `none`} : null} className="catalog__more">
      <button onClick={onMoreFilmsButtonClick} className="catalog__button" type="button">
        Show more
      </button>
    </div>
  );
}

ShowMoreFilms.propTypes = {
  onMoreFilmsButtonClick: PropTypes.func.isRequired,
  filmsToRender: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    genre: PropTypes.string,
    releaseYear: PropTypes.number,
    imgSrc: PropTypes.string,
    bgSrc: PropTypes.string,
    posterSrc: PropTypes.string,
    ratingScore: PropTypes.number,
    ratingCount: PropTypes.number,
    description: PropTypes.arrayOf(PropTypes.string),
    director: PropTypes.string,
    starring: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.number,
    filmDuration: PropTypes.number,
    reviews: PropTypes.array,
  })).isRequired,
  filmsToShowCount: PropTypes.number.isRequired,
};

const mapStateToProps = (state) => ({
  filmsToRender: getFilmsToRender(state),
  filmsToShowCount: getFilmsToShowCount(state)
});


const mapDispatchToProps = (dispatch) => ({
  onMoreFilmsButtonClick: () => {
    dispatch(ActionCreators.showMoreFilms());
  }
});

export {ShowMoreFilms};

export default connect(mapStateToProps, mapDispatchToProps)(ShowMoreFilms);
