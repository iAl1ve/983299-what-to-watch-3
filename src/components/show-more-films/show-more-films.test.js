import React from "react";
import renderer from "react-test-renderer";
import {ShowMoreFilms} from "./show-more-films.jsx";

const mock = {
  filmsList: [
    {
      title: `Some Title`,
      genre: `Comedy`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      description: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 2,
      videoSrc: `Some Path`,
      filmDuration: 99,
      reviews: [
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        }
      ]
    },
    {
      title: `Some Title`,
      genre: `Comedy`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      description: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 4,
      videoSrc: `Some Path`,
      filmDuration: 99,
      reviews: [
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        },
        {
          rating: 8.1,
          reviewText: `Description`,
          reviewer: `Kate Muiry`,
          reviewDate: `2016-12-25`,
        }
      ]
    }
  ]
};

it(`<GenreList /> should render correctly`, () => {
  const tree = renderer
    .create(
        <ShowMoreFilms
          filmsToShowCount={1}
          onMoreFilmsButtonClick={() => {}}
          filmsToRender={mock.filmsList}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
