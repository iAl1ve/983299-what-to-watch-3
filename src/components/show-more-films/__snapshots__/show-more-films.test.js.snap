// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreList /> should render correctly 1`] = `
<div
  className="catalog__more"
  style={null}
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
